plugins/channel_header_call/channel_header_call.jsx

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable react/no-multi-comp */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Dropdown, Tooltip} from 'react-bootstrap';
+import {RootCloseWrapper} from 'react-overlays';
+import {FormattedMessage} from 'react-intl';
+
+import HeaderIconWrapper from 'components/channel_header/components/header_icon_wrapper';
+import PluginChannelHeaderIcon from '../../components/widgets/icons/plugin_channel_header_icon';
+import {Constants} from 'utils/constants';
+import OverlayTrigger from 'components/overlay_trigger';
+
+class CustomMenu extends React.PureComponent {
+    static propTypes = {
+        open: PropTypes.bool,
+        children: PropTypes.node,
+        onClose: PropTypes.func.isRequired,
+        rootCloseEvent: PropTypes.oneOf(['click', 'mousedown']),
+    }
+
+    handleRootClose = () => {
+        this.props.onClose();
+    }
+
+    render() {
+        const {
+            open,
+            rootCloseEvent,
+            children,
+        } = this.props;
+
+        return (
+            <RootCloseWrapper
+                disabled={!open}
+                onRootClose={this.handleRootClose}
+                event={rootCloseEvent}
+            >
+                <ul
+                    role='menu'
+                    className='dropdown-menu channel-header_plugin-dropdown'
+                >
+                    {children}
+                </ul>
+            </RootCloseWrapper>
+        );
+    }
+}
+
+class CustomToggle extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.element,
+        dropdownOpen: PropTypes.bool,
+        onClick: PropTypes.func,
+    }
+
+    handleClick = (e) => {
+        this.props.onClick(e);
+    }
+
+    render() {
+        const {children} = this.props;
+
+        let activeClass = '';
+        if (this.props.dropdownOpen) {
+            activeClass = ' channel-header__icon--active';
+        }
+
+        return (
+            <button
+                id='pluginChannelHeaderButtonDropdown'
+                className={'channel-header__icon channel-header__icon--wide ' + activeClass}
+                type='button'
+                onClick={this.handleClick}
+            >
+                {children}
+            </button>
+        );
+    }
+}
+
+export default class ChannelHeaderCall extends React.PureComponent {
```

Comment:
Similarly, are we able to reuse more of the `ChannelHeaderPlug` here? I haven't looked at the code for `ChannelHeaderPlug`, but most of this looks very generic outside of the icon/tooltip used for the dropdown toggle and what's passed in as `components` by `mapStateToProps`. Perhaps they could use the same component and just pass in props for the things that are different?